csharp/src/Ice/IncomingRequestFrame.cs

```diff
@@ -66,14 +66,15 @@ internal IncomingRequestFrame(Communicator communicator, OutgoingRequestFrame fr
 
         /// <summary>Reads the empty parameter list, calling this methods ensure that the frame payload
         /// correspond to the empty parameter list.</summary>
-        public void ReadEmptyParamList() => InputStream.ReadEmptyEncapsulation(_communicator, Payload);
+        public void ReadEmptyParamList() =>
+            InputStream.ReadEmptyEncapsulation(_communicator, Ice1Definitions.Encoding, Payload);
 
         /// <summary>Reads the request frame parameter list.</summary>
         /// <param name="reader">An InputStreamReader delegate used to read the request frame
         /// parameters.</param>
         /// <returns>The request parameters, when the frame parameter list contains multiple parameters
         /// they must be return as a tuple.</returns>
         public T ReadParamList<T>(InputStreamReader<T> reader) =>
-            InputStream.ReadEncapsulation(_communicator, Payload, reader);
+            InputStream.ReadEncapsulation(_communicator, Ice1Definitions.Encoding, Payload, reader);
```

Comment:
Ditto, I would change `ReadEncapsulation` here to `ReadEncapsulationPayload(_communicator, Encoding, Payload)`.